src/main/java/org/mockito/internal/runners/util/RunnerProvider.java

```diff
@@ -4,24 +4,24 @@
  */
 package org.mockito.internal.runners.util;
 
+import org.mockito.internal.runners.RunnerImpl;
+
 import java.lang.reflect.Constructor;
 import java.lang.reflect.InvocationTargetException;
 
-import org.mockito.internal.runners.RunnerImpl;
-
 public class RunnerProvider {
 
-    public RunnerImpl newInstance(String runnerClassName, Class<?> constructorParam) throws Exception {
+    public RunnerImpl newInstance(String runnerClassName, Object ... constructorArgs) throws Exception {
         Constructor<?> constructor;
         try {
             Class<?> runnerClass = Class.forName(runnerClassName);
-            constructor = runnerClass.getConstructor(Class.class.getClass());
+            constructor = runnerClass.getConstructors()[0];
```

Comment:
Good point! For simplicity I'll just add a check that we expect exactly one constructor.